### ISSUE TITLE
#use_zone doesn't work with Thread.new

### ISSUE THREAD

**Issue by leomao10** [rails/rails#21795]
Hi There,

Please check following codes:
```ruby
# In application.rb
config.time_zone = 'utc'

# run following code in console:
Time.use_zone("Sydney") do
  puts "Outer start #{Thread.current.object_id} with Timezone #{Time.zone.now.zone}"
  Thread.new do
    puts "Future start #{Thread.current.object_id} with Timezone #{Time.zone.now.zone}"
    sleep(3)
    puts "Future finish #{Thread.current.object_id} with Timezone #{Time.zone.now.zone}"
  end
  puts "Outer finish #{Thread.current.object_id} with Timezone #{Time.zone.now.zone}"
end
```

I would expect `Time.zone` within `Thread.new` use the same `Time.zone` with `Time.use_zone` as it is inside `Time.use_zone` block.

But surprisingly, I got following result:
```
# Outer start 70154751181320 with Timezone AEST
# Future start 70154777535080 with Timezone UTC
# Outer finish 70154751181320 with Timezone AEST
# Future finish 70154777535080 with Timezone UTC
```
`Thread.new` doesn't use the timezone I set in `Time.use_zone`, and pick the default time_zone directly.

Also, I found that inside `Thread.new`,  `Thead.current[:time_zone]` is empty.

Shouldn't `Thread.new` copy Thread.current from its outter Thread?

**Comment by sgrif** [comment]
This is the behavior of Ruby, and a side effect of using thread local variables, which is the only way to accomplish this safely. There isn't really a bug nor anything we could do.

**Comment by leomao10** [comment]
I see, thanks for your reply.

Cheers

Leo Liang

On Mon, Sep 28, 2015 at 11:06 PM, Sean Griffin <notifications@github.com>
wrote:

> Closed #21795 <https://github.com/rails/rails/issues/21795>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rails/rails/issues/21795#event-420717160>.
>



-- 
Cheers

Leo Liang
